sync-android/src/main/java/com/cloudant/sync/datastore/encryption/AndroidKeyProvider.java

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright (c) 2015 IBM Cloudant, Inc. All rights reserved.
+ * <p/>
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * <p/>
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+package com.cloudant.sync.datastore.encryption;
+
+import android.content.Context;
+
+import java.util.logging.Logger;
+
+/**
+ * This class implements the interface {@link KeyProvider} and it can be used to create an
+ * encrypted datastore.
+ * <p/>
+ * Given an user-provided password and an identifier, it generates a strong key and store it safely
+ * in the applications {@link android.content.SharedPreferences}, so the same key can be retrieved
```

Comment:
Changes requested in comment included in https://github.com/cloudant/sync-android/commit/dd276d263bfc9c5ddebb047cecd27230cb074f63